include/fmt/colors.h

```diff
@@ -0,0 +1,238 @@
+// Formatting library for C++ - the core API
+//
+// Copyright (c) 2012 - present, Victor Zverovich
+// All rights reserved.
+//
+// For the license information refer to format.h.
+//
+// Copyright (c) 2018 - present, Remotion (Igor Schulz)
+// All Rights Reserved
+// {fmt} support for rgb color output.
+
+#ifndef FMT_COLORS_H_
+#define FMT_COLORS_H_
+
+
+#include "format.h"
+
+FMT_BEGIN_NAMESPACE
+
+// rgb is a struct for red, green and blue colors.
+// we use rgb as name because some editors will show it as color direct in the editor.
+struct rgb {
+  FMT_CONSTEXPR_DECL rgb() : r(0), g(0), b(0) {}
+  FMT_CONSTEXPR_DECL rgb(uint8_t r_, uint8_t g_, uint8_t b_) : r(r_), g(g_), b(b_) {}
+  FMT_CONSTEXPR_DECL rgb(uint32_t hex) 
+      : r((hex >> 16) & 0xFF), g((hex >> 8) & 0xFF), b((hex) & 0xFF) {}
+  uint8_t r;
+  uint8_t g;
+  uint8_t b;
+};
+
+namespace internal {
+
+FMT_CONSTEXPR inline void to_esc(uint8_t c, char out[], int offset) {
+  out[offset + 0] = static_cast<char>('0' + c / 100);
+  out[offset + 1] = static_cast<char>('0' + c / 10 % 10);
+  out[offset + 2] = static_cast<char>('0' + c % 10);
+}
+
+} // namespace internal
+
+FMT_FUNC void vprint_rgb(rgb fd, string_view format, format_args args) {
+  char escape_fd[] = "\x1b[38;2;000;000;000m";
+  static FMT_CONSTEXPR_DECL const char RESET_COLOR[] = "\x1b[0m";
+  internal::to_esc(fd.r, escape_fd, 7);
+  internal::to_esc(fd.g, escape_fd, 11);
+  internal::to_esc(fd.b, escape_fd, 15);
+  
+  std::fputs(escape_fd, stdout);
+  vprint(format, args);
+  std::fputs(RESET_COLOR, stdout);
+}
+
+
+FMT_FUNC void vprint_rgb(rgb fd, rgb bg, string_view format, format_args args) {
+  char escape_fd[] = "\x1b[38;2;000;000;000m"; // foreground color
+  char escape_bg[] = "\x1b[48;2;000;000;000m"; // background color
+  static FMT_CONSTEXPR_DECL const char RESET_COLOR[] = "\x1b[0m";
+  internal::to_esc(fd.r, escape_fd, 7);
+  internal::to_esc(fd.g, escape_fd, 11);
+  internal::to_esc(fd.b, escape_fd, 15);
+  
+  internal::to_esc(bg.r, escape_bg, 7);
+  internal::to_esc(bg.g, escape_bg, 11);
+  internal::to_esc(bg.b, escape_bg, 15);
+  
+  std::fputs(escape_fd, stdout);
+  std::fputs(escape_bg, stdout);
+  vprint(format, args);
+  std::fputs(RESET_COLOR, stdout);
+}
+
+template <typename... Args>
+inline void print_rgb(rgb fd, string_view format_str, const Args & ... args) {
+  vprint_rgb(fd, format_str, make_format_args(args...));
+}
+
+// rgb foreground color
+template <typename... Args>
+inline void print(rgb fd, string_view format_str, const Args & ... args) {
+  vprint_rgb(fd, format_str, make_format_args(args...));
+}
+
+// rgb foreground color and background color
+template <typename... Args>
+inline void print(rgb fd, rgb bg, string_view format_str, const Args & ... args) {
+  vprint_rgb(fd, bg, format_str, make_format_args(args...));
+}
+
+
+enum hex : uint32_t { // use enum class ?
```

Comment:
This would probably be better as an enum class. Also, maybe rename it to something more descriptive, like `color`? I'm also not sure of the CamelCasing of the variants, and if it's consistent with the rest of the library.

Reply:
> This would probably be better as an enum class.
Probably but I really do not want this to be too long.

```cpp
fmt::print(fmt::Coral, "message {}\n", msg);
fmt::print(fmt::colors::Coral, "message {}\n", msg);
```

Also color names are from this list 
https://www.w3schools.com/colors/colors_names.asp

Also Visual Studio and VS Code can directly show colors in the editor using this names.

https://marketplace.visualstudio.com/items?itemName=anseki.vscode-color

https://marketplace.visualstudio.com/items?itemName=lihui.vs-color-picker

Unfortunately I could not find a way to support hex colors like #FFFFFF  in C++.

Reply:
What I meant was more like using `snake_case` instead of `CamelCase` since the library and C++ code in general tends to prefer the former over the latter. But you're right about those extensions, they seem to require the color names to be `lowercase` or `CamelCase`.

Reply:
I agree with @eliaskosunen's points that it should be a enum class `color` with snake case enum constants for consistency with the rest of the code:

```c++
enum class colors : uint32_t {
  alice_blue = ...
};
```